### PR TITLE
don't iterate over star imports

### PR DESCRIPTION
**Pull request by KRRT7**
### **PR Type**
Bug fix, Enhancement, Tests

___

### **Description**
- Skip collecting star imports in collector

- Resolve star imports to concrete symbols

- Add import resolution into add_needed_imports

- Introduce tests for star imports handling

___

### Diagram Walkthrough

```mermaid
flowchart LR
  collector["DottedImportCollector skips star imports"] -- "avoids '*'" --> imports["Accurate import set"]
  resolver["resolve_star_import(module, project_root)"] -- "__all__ or public names" --> symbols["Resolved symbols"]
  addImports["add_needed_imports_from_module"] -- "obj == '*'" --> resolver
  symbols -- "add explicit imports" --> addImports
  tests["New tests"] -- "validate behavior" --> collector
  tests -- "validate behavior" --> resolver
  tests -- "validate behavior" --> addImports
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>code_extractor.py</strong><dd><code>Star import resolution and collector skip</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

codeflash/code_utils/code_extractor.py

<ul><li>Ignore cst.ImportFrom star imports during collection<br> <li> Add resolve_star_import to map '*' to symbols<br> <li> Enhance add_needed_imports to expand star imports<br> <li> Add logging and robust fallbacks</ul>

</details>

  </td>
  <td><a href="https://github.com/codeflash-ai/codeflash/pull/776/files#diff-d76cd127e1798f120f04726e1a3775840a9870e224395fd25bae32235c6982d0">+86/-3</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_add_needed_imports_from_module.py</strong><dd><code>Tests for star import resolution and skipping</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/test_add_needed_imports_from_module.py

<ul><li>Add tests for resolve_star_import (__all__/public)<br> <li> Test non-existent module returns empty set<br> <li> Verify DottedImportCollector skips star imports<br> <li> Test add_needed_imports expands star imports</ul>

</details>

  </td>
  <td><a href="https://github.com/codeflash-ai/codeflash/pull/776/files#diff-761e9ed15c22dc26c56bc95df70d3267258bb83485ac6d76a8f1a0df0ac19c8f">+143/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: tests/test_add_needed_imports_from_module.py [hunk 1]
```diff
@@ -349,3 +353,142 @@ def build_parent_map(self, nodes: Dict, base: Optional[bool] = False) -> Dict[st
...
+        src_path = project_root / 'src.py'
+        dst_path = project_root / 'dst.py'
+        src_path.write_text(src_code)
+        
+        result = add_needed_imports_from_module(
+            src_code, dst_code, src_path, dst_path, project_root
+        )
+        
+        # The result should have individual imports instead of star import
+        assert 'from source_module import' in result
```
Comment (misrasaurabh1) [inline tests/test_add_needed_imports_from_module.py]: this test looks incomplete...

**Review by misrasaurabh1 — APPROVED** [review]

**Review by misrasaurabh1 — APPROVED** [review]
